pyccel/decorators.py

```diff
@@ -130,6 +130,40 @@ def kernel(f):
     KernelAccessor
         A class representing the kernel function.
     """
+    class CudaThreadIndexing:
+        """
+        Class representing the CUDA thread indexing.
+
+        Class representing the CUDA thread indexing.
+        """
+        def __init__(self, block_idx, thread_idx):
+            self._block_idx = block_idx
+            self._thread_idx = thread_idx
+
+        def threadIdx(self, dim):
+            """
+            Get the thread index.
+
+            Get the thread index.
+            """
+            return self._thread_idx
+
+        def blockIdx(self, dim):
```

Comment:
why not implement the logic for the dim argument from the start?

Reply:
For the moment, you can't use tuples for num_blocks and num_threads. If you pass these values as tuples to the kernel call, you will receive a 'not implemented' error. Once this issue is resolved, I will implement the logic for the dim